src/parser/DASHTree.cpp

```diff
@@ -583,7 +583,10 @@ void adaptive::CDashTree::ParseTagAdaptationSet(pugi::xml_node nodeAdp, PLAYLIST
   xml_node nodeSegTpl = nodeAdp.child("SegmentTemplate");
   if (nodeSegTpl || period->HasSegmentTemplate())
   {
-    CSegmentTemplate segTemplate{period->GetSegmentTemplate()};
+    CSegmentTemplate segTemplate;
+    auto optSegTemplate = period->GetSegmentTemplate();
+    if (optSegTemplate)
+      segTemplate = std::move(*optSegTemplate);
```

Comment:
the initial intention about using constructor was to avoid all this verbose code, that result also repeated multiple times for each use

can you try add an assign operator method to the classes?
`CSegmentTemplate& operator=(const std::optional<CSegmentTemplate>& other)`
then should result as:
```
CSegmentTemplate segTemplate;
segTemplate = period->GetSegmentTemplate();
```
and see if compiler still cause problems?

Reply:
That should work I think. But what are you opinions on the version I just pushed? Shorter and it makes it obvious that there maybe wasn't a value returned.

Reply:
very good! i didn't remember there was a builtin method, i will do a test soon as soon as i find some time